packages/ember-data/lib/system/model/states.js

```diff
@@ -329,7 +326,7 @@ var DirtyState = {
     },
 
     didSetProperty: function(record, context) {
-      get(record, 'errors').remove(context.name);
+      record.getErrors().remove(context.name);
```

Comment:
i know these are probably just some methods moved around, but it feels weird to mutate another object's property's state. perhaps we could call record.removeError(context.name) ?